pkg/build/metrics/prometheus/metrics.go

```diff
@@ -77,40 +96,58 @@ func (bc *buildCollector) Collect(ch chan<- prometheus.Metric) {
 	// to be added to the total amount posted to prometheus
-	var failed, error, cancelled, complete int
+	var error, cancelled, complete int
+	reasons := map[string]int{}
 	for _, b := range result {
-		f, e, cc, cp := bc.collectBuild(ch, b)
-		failed = failed + f
+		e, cc, cp, r := bc.collectBuild(ch, b)
+		for key, value := range r {
+			count := reasons[key]
+			count = count + value
+			reasons[key] = count
+		}
 		error = error + e
 		cancelled = cancelled + cc
 		complete = complete + cp
 	}
-	addCountGauge(ch, terminalBuildCountDesc, failedPhase, float64(failed))
+	// explicitly note there are no failed builds
+	if len(reasons) == 0 {
+		addCountGauge(ch, failedBuildCountDesc, "", float64(0))
+	}
+	for reason, count := range reasons {
+		addCountGauge(ch, failedBuildCountDesc, reason, float64(count))
+	}
 	addCountGauge(ch, terminalBuildCountDesc, errorPhase, float64(error))
 	addCountGauge(ch, terminalBuildCountDesc, cancelledPhase, float64(cancelled))
 	addCountGauge(ch, terminalBuildCountDesc, completePhase, float64(complete))
 }
 
-func addCountGauge(ch chan<- prometheus.Metric, desc *prometheus.Desc, phase string, v float64) {
-	lv := []string{phase}
+func addCountGauge(ch chan<- prometheus.Metric, desc *prometheus.Desc, label string, v float64) {
+	lv := []string{label}
 	ch <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, v, lv...)
 }
 
-func addTimeGauge(ch chan<- prometheus.Metric, b *buildapi.Build, desc *prometheus.Desc) {
-	if b.Status.StartTimestamp != nil {
-		lv := []string{b.ObjectMeta.Namespace, b.ObjectMeta.Name, strings.ToLower(string(b.Status.Phase))}
-		ch <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, float64(b.Status.StartTimestamp.Unix()), lv...)
+func addTimeGauge(ch chan<- prometheus.Metric, b *buildapi.Build, time *metav1.Time, desc *prometheus.Desc, phase string) {
+	if time != nil {
+		lv := []string{b.ObjectMeta.Namespace, b.ObjectMeta.Name}
+		if len(phase) > 0 {
+			lv = append(lv, strings.ToLower(phase))
+		}
+		ch <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, float64(time.Unix()), lv...)
 	}
 }
 
-func (bc *buildCollector) collectBuild(ch chan<- prometheus.Metric, b *buildapi.Build) (failed, error, cancelled, complete int) {
+func (bc *buildCollector) collectBuild(ch chan<- prometheus.Metric, b *buildapi.Build) (error, cancelled, complete int, reasons map[string]int) {
 
 	switch b.Status.Phase {
 	// remember, new and pending builds don't have a start time
+	case buildapi.BuildPhaseNew:
+	case buildapi.BuildPhasePending:
+		addTimeGauge(ch, b, &b.CreationTimestamp, newPendingBuildCountDesc, string(b.Status.Phase))
 	case buildapi.BuildPhaseRunning:
-		addTimeGauge(ch, b, activeBuildCountDesc)
+		addTimeGauge(ch, b, b.Status.StartTimestamp, activeBuildCountDesc, "")
 	case buildapi.BuildPhaseFailed:
-		failed++
+		// currently only failed builds have reasons
+		reasons[string(b.Status.Reason)] = 1
 	case buildapi.BuildPhaseError:
 		error++
```

Comment:
you may live to regret this variable name choice.

Reply:
ah ... a golang thing perhaps ... good catch / I'll change